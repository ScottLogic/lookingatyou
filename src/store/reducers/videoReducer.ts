import {
    IVideo,
    IVideoState,
    SET_IMAGE_DATA,
    SET_VIDEO,
    SET_VIDEO_STREAMS,
    VideoActionTypes,
} from '../actions/video/types';

export const initialState: IVideoState = {
    videos: {},
    images: {},
};

const videoStore = (
    state: IVideoState = initialState,
    action: VideoActionTypes,
): IVideoState => {
    switch (action.type) {
        case SET_VIDEO_STREAMS:
            const newVideos: { [deviceId: string]: IVideo } = {};
            action.videos.map(
                (video: IVideo) => (newVideos[video.deviceId] = video),
            );
            return {
                ...state,
                videos: {
                    ...state.videos,
                    ...newVideos,
                },
            };
        case SET_VIDEO:
            const elementId = action.payload.deviceId;
            const updatedObject: { [key: string]: IVideo } = {};
            updatedObject[elementId] = {
                ...state.videos[elementId],
                ...action.payload,
            };
            return {
                ...state,
                videos: {
                    ...state.videos,
                    ...updatedObject,
                },
            };
        case SET_IMAGE_DATA:
            return {
                ...state,
                images: {
                    ...state.images,
                    ...action.images,
                },
            };
        default:
            return state;
    }
};

export default videoStore;
