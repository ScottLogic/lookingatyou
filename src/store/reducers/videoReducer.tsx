import {
    IVideo,
    IVideoState,
    SET_VIDEO,
    SET_VIDEO_STREAMS,
    VideoActionTypes,
} from '../actions/video/types';

export const initialState: IVideoState = {
    videos: {},
};

const videoStore = (
    state: IVideoState = initialState,
    action: VideoActionTypes,
): IVideoState => {
    switch (action.type) {
        case SET_VIDEO_STREAMS:
            const videos: { [deviceId: string]: IVideo } = {};
            action.videos.map(
                (video: IVideo) => (videos[video.deviceId] = video),
            );
            return { ...state, videos };
        case SET_VIDEO:
            return {
                ...state,
                ...(state.videos[action.payload.deviceId].video =
                    action.payload.video),
            };
        default:
            return state;
    }
};

export default videoStore;
