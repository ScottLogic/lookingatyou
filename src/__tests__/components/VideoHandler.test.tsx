import { shallow } from 'enzyme';
import React from 'react';
import {
    VideoHandler,
    VideoHandlerProps,
} from '../../components/video/VideoHandler';
import { testDevice1, testDevice2 } from '../reducers/video/videoReducer.test';

let props: VideoHandlerProps;

describe('Video handler', () => {
    beforeEach(() => {
        props = {
            mediaDevices: null,
            deviceIds: [testDevice1, testDevice2],
            configureStream: jest.fn(),
        };
    });

    it('Component should render correctly', () => {
        const wrapper = shallow(<VideoHandler {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });
});
