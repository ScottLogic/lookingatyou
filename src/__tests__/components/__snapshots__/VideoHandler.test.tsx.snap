// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video handler Component should render correctly 1`] = `
"<div className=\\"webcam-feed\\">
  <Connect(Video) deviceId=\\"testDevice1\\" />
  <Connect(Video) deviceId=\\"testDevice2\\" />
</div>"
`;
